### ISSUE TITLE
UDP broadcast port 8086 reserved on Apple systems?

### ISSUE THREAD

**Issue by slomp** [wolfpld/tracy#625]
I ran into issue on macOS where Tracy profiler was unable to listen for UDP broadcast messages from clients on port 8086. More specifically, socket `bind()` would fail with "error 48 address in use" even though the address reuse socket option (`SO_REUSEADDR`) was enabled.

I found out that port 8086 was being used by the system for something, and that I could not even kill the related process. Rebooting the system did not help at first, but on the next day, after another reboot, port 8086 was free again. I am not sure what prompts the system to star using the port.

After some digging, it looks like Apple has the range of ports [8000, 8999] reserved for "Web service, iTunes Radio streams", whatever that means:
https://support.apple.com/en-us/HT202944

In addition, port 8086 is also an officially assigned IANA port as of 2008 for "Distributed SCADA Networking Rendezvous Port":
https://www.adminsub.net/tcp-udp-port-finder/8086
https://www.speedguide.net/port.php?port=8086
https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xml

If we were to stick with the Intel homage, perhaps we could use port 4004, which does not seem to be used for anything "mainstream".

**Comment by wolfpld** [comment]
With the amount of software available, each port is sure to be used by several different programs, and it's no longer possible to assign a unique port for yourself for all time.

I'm not sure that fixing a local problem (something hogging a port) by changing the configuration for everyone is the right solution here. I would expect at least a number of other people to confirm that they have the same problem.

My main concern is that right now you can run the latest version of the profiler and it will tell you which version of the profiler you need to download if you are using an older client. If you were to change the listen port, you would no longer get the broadcasts that older clients send.

**Comment by slomp** [comment]
Yeah, that's a fair point -- even knowing that port 8086 is technically "off-limits" on Apple platforms.

Perhaps a good action plan would be to print a warning in red in the profiler window when ` UdpListen::Listen()` fails.
This will make it easier for people to spot issues and report back (as it stands, it silently fails with no report).

**Comment by wolfpld** [comment]
Yeah, notifying the user would be good.
